eclair-core/src/main/scala/fr/acinq/eclair/router/Validation.scala

```diff
@@ -275,12 +275,18 @@ object Validation {
       // related channel is already known (note: this means no related channel_update is in the stash)
       val publicChannel = true
       val pc = d.channels(u.shortChannelId)
+      val pc1 = update match {
+        case Left(lcu) => pc.updateChannelUpdateSameSideAs(u).updateBalances(lcu.commitments)
+        case Right(_) => pc.updateChannelUpdateSameSideAs(u)
+      }
       val desc = getDesc(u, pc.ann)
```

Comment:
```suggestion
      val desc = getDesc(u, pc1.ann)
```

And all the instances below, we should be using `pc1` and not `pc` anymore.

Reply:
Actually no, on the contrary, the usage below needs to use `pc` because it checks the previous state compare to the update we're receiving.
For the line you suggest it ends up being the same, but I think using `pc` makes more sense.

Reply:
On the contrary, the usage below needs to use `pc` because it checks the previous state compare to the update we're receiving.
For the line you suggest it ends up being the same, but I think using `pc` makes more sense.

Reply:
I see, i was confused by the naming, i think it's not clear because we usually use this pattern to update an object and use the updated instance from there on in the code. We should rename those `pc` and `pc1` to something more readable, but probably in another PR.